sunpy/coordinates/spice.py

```diff
@@ -467,3 +467,65 @@ def get_fov(instrument, time, *, resolution=100):
                    obstime=obstime,
                    representation_type='unitspherical')
     return fov
+
+def get_rotation_matrix(source_frame, target_frame, from_time, to_time=None):
+    """
+    Computes the rotation matrix that converts vectors from the `source_frame` to the `target_frame`
+    at the specified time using the SPICE toolkit.
+
+    Parameters
+    ----------
+    source_frame : `str`, `int`, `~astropy.coordinates.BaseCoordinateFrame`
+        The frame of the input vector field, specified as a frame name, SPICE ID, or Astropy BaseCoordinateFrame.
+
+    target_frame : `str`, `int`, `~astropy.coordinates.BaseCoordinateFrame`
+        The frame to which the vector field will be transformed, specified in the same formats as ``source_frame``.
+
+    from_time : `str`, `~astropy.time.Time`, `~datetime.datetime`, `~datetime.date`, `~numpy.datetime64`, `~pandas.Series`, `~pandas.DatetimeIndex`, `~pandas.DataFrame`
+        The time at which the vector field is defined in the source frame, in any format accepted by ``sunpy.time.parse_time``.
+
+    to_time : `str`, `~astropy.time.Time`, `~datetime.datetime`, `~datetime.date`, `~numpy.datetime64`, `~pandas.Series`, `~pandas.DatetimeIndex`, `~pandas.DataFrame`, optional
+        The time at which the vector field should be defined in the target frame. Defaults to ``from_time``, resulting in a spatial-only transformation.
+
+    Returns
+    -------
+    `~numpy.ndarray`
+        A 3x3 rotation matrix that represents the rotation between the two frames.
+
+    Examples
+    --------
+    >>> from sunpy.coordinates.spice import get_rotation_matrix
+    >>> source_frame = "J2000"
+    >>> target_frame = "Galactic"
+    >>> from_time = '2001-01-01T00:00:00'
+
+    >>> rotation_matrix = get_rotation_matrix(source_frame, target_frame, from_time)
+
+    * The `get_rotation_matrix` function will first determine the rotation matrix that converts vectors from the source frame to the target frame at the specified time.
+    * This rotation matrix is a 3x3 matrix that represents the rotation between the two frames.
+
+    >>> rotation_matrix
+    array([[ 0.999, -0.034,  0.002],
+           [ 0.034,  0.999,  0.004],
+           [-0.002, -0.004,  1.000]])
+
+    * `rotation_matrix` can now be used with the vector components to transform the field.
```

Comment:
Should we have an example with the `@` after this?

Reply:
Yeah sure, that might be useful